theano/tensor/tests/test_opt.py

```diff
@@ -1713,6 +1714,27 @@ def test_local_useless_subtensor():
         f([[1, 2, 3], [4, 5, 6]], 1)
         f([[1, 2, 3], [4, 5, 6]], 3)
 
+    # Test AdvancedSubtensor1 case when all rows are selected by a list/vector
+    # or ARange op
+    for dims, res in (([0, 1], True),
+                      ([1, 0], False),
+                      ([0, 0], False),
+                      ([0, 0, 1], False),
+                      (T.arange(2), True),
+                      (T.arange(2, -1), False),
+                      (T.arange(1, 2), False)):
```

Comment:
Can you test T.arange(0, 2) and T.arange(0, 2, 2). Just to test the step parameter.